templates/_entries/search/default.twig

```diff
@@ -1,17 +1,114 @@
+{% set searchQuery = craft.app.request.getParam('q') %}
+{% set searchSection = craft.app.request.getParam('section') %}
+{% set searchUpdatedDate = craft.app.request.getParam('updatedDate') %}
+{% set searchWhere = craft.app.request.getParam('searchWhere') %}
+
+{# Date for search select option list #}
+{% set yesturdayDate = "now"|date_modify("-1 day")|date('Y-m-d') %}
+{% set pastWeekDate = "now"|date_modify("-7 day")|date('Y-m-d') %}
+{% set pastMonthDate = "now"|date_modify("-1 month")|date('Y-m-d') %}
+{% set pastYearDate = "now"|date_modify("-1 year")|date('Y-m-d') %}
+
+{# Check if advanced search panel need to be open or closed #}
+{% set openAdvancedSearch = false %}
+{% if searchSection or searchUpdatedDate or searchWhere %}
+    {% set openAdvancedSearch = true %}
+{% endif %}
+
+{# Configuration of the search URL and label that need to be use #}
+{% set searchAction, searchLabel = craft.entries.section('search').one().url() ?? null, 'Search this website'|t %}
+{% if headerSettings.customSearchQueryUrl|length %}
+    {% set searchLabel = 'Search Canada.ca'|t %}
+    {% set searchAction = headerSettings.customSearchQueryUrl %}
+{% endif %}
+
+{# Sections select option #}
+{%  set allSections = craft.query()
```

Comment:
This will return all the sections available. You will need to filter down to actual content page sections.